KSR_002/KSR_002.md

```diff
@@ -0,0 +1,282 @@
+
+**Conclusion - HEAP**: hence we can assume that from a safety point of view, the heap section is not relevant at all.
+
+    
+### [vDSO and vsyscalls ](https://0xax.gitbooks.io/linux-insides/content/SysCall/linux-syscall-3.html):
+vsyscalls are optimized syscalls whose handler implementation is directly mapped statically into the process address space.  
+On x86 (`grep __vsyscall_page ./ -R` in arch/x86) the only vsyscalls are:
+
+- [gettimeofday](https://github.com/torvalds/linux/blob/v5.4/arch/x86/entry/vsyscall/vsyscall_emu_64.S#L20)
+- [time](https://github.com/torvalds/linux/blob/v5.4/arch/x86/entry/vsyscall/vsyscall_emu_64.S#L25)
+- [getcpu](https://github.com/torvalds/linux/blob/v5.4/arch/x86/entry/vsyscall/vsyscall_emu_64.S#L30)
+
+**Conclusion: vsyscalls** are not relevant for the safety app as they are not used at all by the safety app
+
+
+vDSO are the evolution of vsyscalls; the vDSO routines are dynamically linked in the process address space (the object file is `linux-vdso.so.1`).
+On x86 (“grep __vdso_ ./ -R” in arch/x86) the only [vdso syscalls](https://github.com/torvalds/linux/blob/v5.4/arch/x86/entry/vdso/vdso.lds.S#L17) are:
+
+- clock_gettime;
+- gettimeofday;
+- getcpu;
+- time;
+- clock_getres;
+
+**Conclusion: vdso** functions are not relevant for the safety app as they are not used at all by the safety app
+
+**Conclusion: vvar (vdso variable) **are not relevant for the safety app as VDSO functions are not used by the safety app
+
+<mark > **TODO: to formally confirm this, eventually using the calltree tool (or cscope tool)** </mark>
+
+The [modified safety app](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c) code has been annotated to mark the code that is optional (i.e. can be removed without affecting the functionalities of the safety app itself) or that is not active during	the safe operation. After the annotations the following glibc calls are left to be analyzed:
+
+- [mkfifo()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L70): looking at the glibc source code the call-tree is quite simple:  [__mknod](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/mkfifo.c;h=5f9452ce497a6bbd6bc42ea23c4c4af7118c2e09;hb=HEAD#l27)()->__mknod() ->mknodat(). Hence here there is not involvement at all of vdso and vvar
+- [open()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L71): glibc implements a wrapper for the respective kernel syscall
+- [ioctl()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L79): glibc implements a wrapper for the respective kernel syscall
+- [usleep()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L109): 
+
+<table>
+  <tr>
+   <td>
+usleep
+   </td>
+   <td>https://elixir.bootlin.com/glibc/glibc-2.35/source/sysdeps/posix/usleep.c#L23
```

Comment:
All elixir links must be replaced by official packg git repo. E.g. this one would be https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/usleep.c;h=cfb9db1505c4f07ceb025b14e6bb801e50babae1;hb=bc6c1af537dbcc05e1559e02f0f12942a829c224#l23